src/common/base_client.ts

```diff
@@ -401,4 +411,78 @@ export abstract class BaseClient {
             await this.userSubscribe(this.userSubscribeParams);
         }
     }
+
+    private getRequestError(code: number, data?: any) : KolibriRequestError {
```

Comment:
This code can be moved to a separate class.

Reply:
Maybe have it in as a static method in KolibriRequestError?